components/wpa_supplicant/esp_supplicant/src/esp_dpp.c

```diff
@@ -180,12 +180,12 @@ static int esp_dpp_handle_config_obj(struct dpp_authentication *auth,
     wifi_config_t *wifi_cfg = &s_dpp_ctx.wifi_cfg;
 
     if (conf->ssid_len) {
-        os_memcpy(wifi_cfg->sta.ssid, conf->ssid, conf->ssid_len);
+        os_strlcpy((char*)wifi_cfg->sta.ssid, (char*)conf->ssid, sizeof(wifi_cfg->sta.ssid));
     }
 
     if (dpp_akm_legacy(conf->akm)) {
         if (conf->passphrase[0])
-            os_memcpy(wifi_cfg->sta.password, conf->passphrase,
+            os_strlcpy((char*)wifi_cfg->sta.password, (char*)conf->passphrase,
```

Comment:
This has a subtle bug in that if your first start_listen sequence yields a passphrase and your second one does not, you'll technically yield a wifi_config_t with a passphrase from the previous listen.  I think the fix is overall more consistent if you wipe the wifi_cfg data structure on each call to start_listen.  Same bug below with `pmf_cfg.required = true`.

Reply:
Actually there are other existing bugs in this code that suggest wiping s_dpp_ctx.wifi_cfg on each invocation of esp_dpp_handle_config_obj is better.  The caller even runs this function in a loop which seems clearly broken to me...

Reply:
Ah, I must've missed this thanks to the two networks I tested with having the same passphrase, I'll have to look at wiping wifi_cfg. ty